src/services/commit-message/CommitMessageProvider.ts

```diff
@@ -66,17 +67,21 @@ export class CommitMessageProvider {
 				try {
 					progress.report({ increment: 25, message: t("kilocode:commitMessage.analyzingChanges") })
 
-					const changes = await this.gitService.gatherStagedChanges()
-					if (changes === null) {
-						vscode.window.showInformationMessage(t("kilocode:commitMessage.noStagedChangesRepo"))
-						return
-					}
-					if (changes.length === 0) {
-						vscode.window.showInformationMessage(t("kilocode:commitMessage.noStagedChanges"))
-						return
+					let staged = true
+					let changes = await this.gitService.gatherChanges({ staged })
+
+					if (!changes || changes.length === 0) {
+						staged = false
+						changes = await this.gitService.gatherChanges({ staged })
+						if (!changes || changes.length === 0) {
```

Comment:
```suggestion
						if (!changes) {
```

Isn't the second check superfluous?

Reply:
It's a empty array vs null thing, but I think I could clean this up by updating `gitService.gatherChanges` to always be empty array